Sources/System/FilePath/FilePathParsing.swift

```diff
@@ -0,0 +1,112 @@
+/*
+ This source file is part of the Swift System open source project
+
+ Copyright (c) 2020 Apple Inc. and the Swift System project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+*/
+
+private typealias SystemChar = SystemString.Element
+
+// The separator we use internally
+private var genericSeparator: SystemChar {
+  numericCast(UInt8(ascii: "/"))
+}
+
+// The platform preferred separator
+//
+// TODO: Make private
+private var platformSeparator: SystemChar {
+#if os(Windows)
+  return numericCast(UInt8(ascii: "\\"))
+#else
+  return genericSeparator
+#endif
+}
+
+// TODO: Make private
+private func isSeparator(_ c: SystemChar) -> Bool {
+  c == genericSeparator || c == platformSeparator
+}
+
+// @available(macOS 10.16, iOS 14.0, watchOS 7.0, tvOS 14.0, *)
```

Comment:
```suggestion
```

Reply:
Since none of this is public I don't think we need (commented out) availability? Since this is new code, I'm not sure this availability annotation would be accurate?

Reply:
It is necessary due to a compiler limitation where non-public code that uses a public type needs availability. But we can defer adding that until it's relevant to an ABI release